### PR TITLE
Add DropVirtualChannel for DataCoord

### PR DESCRIPTION
**Pull request by congqixia**
Add `DropVirtualChannel` for DataCoord, which is the "Flush&Drop" api 
Detailed see also #12358

/kind enhancement

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/datacoord/meta.go [hunk 1]
```diff
@@ -334,6 +335,136 @@ func (m *meta) UpdateFlushSegmentsInfo(
 	return nil
 }
 
+// UpdateDropChannelSegmentInfo updates segment checkpoints and binlogs before drop
+// reusing segment info to pass segment id, binlogs, statslog, deltalog, start position and checkpoint
+func (m *meta) UpdateDropChannelSegmentInfo(channel string, segments []*SegmentInfo) error {
```
Comment (sunby) [inline internal/datacoord/meta.go:340]: This function is too long, we can split it to multi functions to improve readability.

**Review by sunby — COMMENTED** [review]
If datacoord crash after setting removeFlag, we should remove the channel after it retarts.

**Comment by congqixia** [comment]
> If datacoord crash after set removeFlag, we should remove the channel after it retarts.

@sunby true, this part is related to `ServerHandler.CheckShouldDropChannel`, which should be modified after `DataNode` logic change.

File: internal/datacoord/meta.go (hunk 1 above)
Comment (congqixia) [reply internal/datacoord/meta.go:340]: Fixed

**Pull request merged**